cmd/kubeadm/app/cmd/phases/init/uploadconfig.go

```diff
@@ -103,42 +103,42 @@ func getUploadConfigPhaseFlags() []string {
 }
 
 // runUploadKubeadmConfig uploads the kubeadm configuration to a ConfigMap
-func runUploadKubeadmConfig(c workflow.RunData) error {
-	cfg, client, _, err := getUploadConfigData(c)
+func runUploadKubeadmConfig(c workflow.RunData, phase string) error {
+	cfg, client, _, err := getUploadConfigData(c, phase)
```

Comment:
The phase here can be `kubelet`, `kubeadm`, and `all`.

```
kubeconfig                   Generate all kubeconfig files necessary to establish the control plane and the admin kubeconfig file
  /kubelet                     Generate a kubeconfig file for the kubelet to use *only* for cluster bootstrapping purposes
...
upload-config                Upload the kubeadm and kubelet configuration to a ConfigMap
  /kubeadm                     Upload the kubeadm ClusterConfiguration to a ConfigMap
  /kubelet                     Upload the kubelet component config to a ConfigMap
```

There will be two `[kubelet]`, one for kubeconfig and one for upload-config.

Similarly, `apiserver` and `controller-manager `, `scheduler` also may be for different phase.

```
certs                        Certificate generation
  /apiserver                   Generate the certificate for serving the Kubernetes API
kubeconfig                   Generate all kubeconfig files necessary to establish the control plane and the admin kubeconfig file
  /kubelet                     Generate a kubeconfig file for the kubelet to use *only* for cluster bootstrapping purposes
  /controller-manager          Generate a kubeconfig file for the controller manager to use
  /scheduler                   Generate a kubeconfig file for the scheduler to use
control-plane                Generate all static Pod manifest files necessary to establish the control plane
  /apiserver                   Generates the kube-apiserver static Pod manifest
  /controller-manager          Generates the kube-controller-manager static Pod manifest
  /scheduler                   Generates the kube-scheduler static Pod manifest
upload-config                Upload the kubeadm and kubelet configuration to a ConfigMap
  /kubelet                     Upload the kubelet component config to a ConfigMap
```

Reply:
I'd hope to make the log as precise as possible with minimal cost, your comments here is valid and nice catch, but I don't make it over-complicated, thoughts?

Reply:
- Before, only `parent-phase` will be shown. The log details may show what phase it is.
- After this change, the `sub-phase` is a priority over the `parent-phase`. It is more precise but sometimes dup with another sub-phase.

For instance, before
```
[certs] Using certificateDir folder "/etc/kubernetes/tmp/kubeadm-init-dryrun443308956"
[certs] Using existing ca certificate authority
[certs] Using existing apiserver certificate and key on disk
[certs] Using existing apiserver-kubelet-client certificate and key on disk
[certs] Using existing front-proxy-ca certificate authority
[certs] Using existing front-proxy-client certificate and key on disk
[certs] Using existing etcd/ca certificate authority
[certs] Using existing etcd/server certificate and key on disk
[certs] Using existing etcd/peer certificate and key on disk
[certs] Using existing etcd/healthcheck-client certificate and key on disk
[certs] Using existing apiserver-etcd-client certificate and key on disk
[certs] Generating "sa" key and public key
```
after 
```
[certs] Using certificateDir folder "/etc/kubernetes/tmp/kubeadm-init-dryrun443308956"
[ca] Using existing ca certificate authority
[apiserver] Using existing apiserver certificate and key on disk
[apiserver-kubelet-client] Using existing apiserver-kubelet-client certificate and key on disk
[front-proxy-ca] Using existing front-proxy-ca certificate authority
[front-proxy-client] Using existing front-proxy-client certificate and key on disk
[etcd-ca] Using existing etcd/ca certificate authority
[etcd-server] Using existing etcd/server certificate and key on disk
[etcd-peer] Using existing etcd/peer certificate and key on disk
[etcd-healthcheck-client] Using existing etcd/healthcheck-client certificate and key on disk
[apiserver-etcd-client] Using existing apiserver-etcd-client certificate and key on disk
[ca] Generating "sa" key and public key
```


I prefer the current style in this usecase.

Reply:
> I prefer the current style in this usecase.

+1 i agree. we should try to keep it simple. if some `[  ... ]` prefixes need adjustment we can fix them, though.

Reply:
Also prefer the keep original style here. There is no need to print the specific stages as the prefix, and from the content we can determine what we are doing.